lib/publiccloud/azure.pm

```diff
@@ -82,7 +83,7 @@ sub vault_create_credentials {
 
 sub resource_exist {
     my ($self) = @_;
-    my $output = script_output(q(az group list --query "[?name=='openqa-upload']"));
+    my $output = script_output_retry(q(az group list --query "[?name=='openqa-upload']"), retry => 3, timeout => 30, delay => 10);
```

Comment:
General question, did you see the behavior that it retrieve you nothing but later you got the resource group?

Reply:
ok I see now. this is the first command you are calling, but why not just `$provider->az_login()`. Afterwards the commands should work in the first phase.

Reply:
Ok we do :) :+1: 
Then it get even more worst here :/
Maybe we should do this then directly inside of `az_login()` I mean, if we cannot trust on these commands, we would need to add all `az` calls in such retry wrappers.

Reply:
And there is a other error, we should replace `openqa-upload` with `$self->resource_group`.

Reply:
> What is the difference between `script_retry()` and `script_output_retry()` as both support `die => 0`?

`script_retry` returns the return code, `script_output_retry` returns the output of the script

Reply:
I updated the `$self->resource_group`. Since this a rather needed fix because the runs are failing I would for now move forward and do the `az_login` in another PR. What do you think?